src/operators/_predicates.ts

```diff
@@ -217,25 +217,40 @@ export function $exists(a: AnyVal, b: AnyVal, options?: Options): boolean {
 /**
  * Matches arrays that contain all elements specified in the query.
  *
- * @param a
- * @param b
+ * @param values
+ * @param queries
  * @returns boolean
  */
-export function $all(a: RawArray, b: RawArray, options?: Options): boolean {
-  let matched = false;
-  if (isArray(a) && isArray(b)) {
-    for (let i = 0, len = b.length; i < len; i++) {
-      if (isObject(b[i]) && inArray(Object.keys(b[i]), "$elemMatch")) {
-        matched = matched || $elemMatch(a, b[i]["$elemMatch"], options);
+export function $all(
+  values: RawArray,
+  queries: RawArray,
+  options?: Options
+): boolean {
+  let matched = true;
+  if (isArray(values) && isArray(queries)) {
+    for (let i = 0, len = queries.length; i < len; i++) {
+      const query = queries[i];
+      if (isObject(query) && inArray(Object.keys(query), "$elemMatch")) {
+        matched = matched && $elemMatch(values, query["$elemMatch"], options);
+      } else if (query instanceof RegExp) {
+        matched =
+          matched &&
+          values.some((value) => testValueAgainstRegExp(value, query));
       } else {
-        // order of arguments matter
-        return intersection(b, a, options?.hashFunction).length === len;
```

Comment:
The problem is that I can't use intersection as before because I am now comparing a single query item with a list of values, whereas it previously assumed that we are comparing both arrays. What is your recommended approach to enable a deep equality comparison? I could use a utility function from `lodash`?

Reply:
There is a deep equality check function [isEqual](https://github.com/kofrasa/mingo/blob/master/src/util.ts#L352) from the `util` module that you can use.

Reply:
Alright, I have pushed up the change to is `isEqual`

Reply:
Thanks for the update. I just happen to have some time early today and incorporated your first changeset (with the updates) before seeing this.

Thanks again for the contribution. I very much appreciate it.

Reply:
I added an additional test for the deep equals case, if you want to port it over: https://github.com/kofrasa/mingo/pull/189/files#diff-124b6fc8a136b24fdfb40d619f7e455c5f4f923be442393e1612f4d3e6f0e3c7R278